tests/testthat/test-bind.R

```diff
@@ -267,6 +267,73 @@ test_that("can assign row names in vec_rbind()", {
   expect_identical(out, exp)
 })
 
+test_that("vec_rbind() takes the proxy and restores", {
+  df <- foobar(data.frame(x = 1))
+
+  # This data frame subclass has an identity proxy and the restore
+  # method falls back to a bare data frame if `$x` has any missing values.
+  # In `vec_rbind()`, the `vec_init()` call will create a bare data frame,
+  # but at the end it is `vec_restore()`d to the right class.
+  local_methods(
+    vec_ptype2.vctrs_foobar.vctrs_foobar = function(x, y, ...) {
+      x
+    },
+    vec_proxy.vctrs_foobar = function(x, ...) {
+      x
+    },
+    vec_restore.vctrs_foobar = function(x, to, ...) {
+      if (any(is.na(x$x))) {
+        new_data_frame(x)
+      } else {
+        NextMethod()
```

Comment:
Can you make these tests work without calling next-method? The semantics of default methods are likely to change in 0.4.0.

Reply:
I could directly call `vec_restore_default()`?